src/effects/effectparameter.cpp

```diff
@@ -2,14 +2,12 @@
 
 #include "effects/effectparameter.h"
 #include "effects/effectsmanager.h"
-#include "effects/effect.h"
 #include "util/assert.h"
 
-EffectParameter::EffectParameter(Effect* pEffect, EffectsManager* pEffectsManager,
-                                 int iParameterNumber,
-                                 EffectManifestParameterPointer pParameter)
+EffectParameter::EffectParameter(EffectSlot* pEffectSlot, EffectsManager* pEffectsManager,
+                                 int iParameterNumber, EffectManifestParameterPointer pParameter)
         : QObject(), // no parent
-          m_pEffect(pEffect),
+          m_pEffectSlot(pEffectSlot),
```

Comment:
Change this to an `EngineEffect*`, then you can remove EffectSlot::getEngineEffect